udf/udf.proto

```diff
@@ -0,0 +1,183 @@
+syntax = "proto3";
+
+package udf;
+
+
+//------------------------------------------------------
+// RPC Messages for Kapacitor to communicate with
+// a child process for data processing.
+//
+// The Kapacitor processes will send Request messages
+// over STDIN to the child process.
+// The child process can respond over STDOUT.
+//
+// Messages are streamed by writing a varint header
+// that contains the length of the following message.
+// 
+// To decode the stream read a varint, then read
+// the determined size and decode as a protobuf message.
+// There is not footer so the next varint if any begins
+// right after the previous message.
+//
+
+//------------------------------------------------------
+// Management messages
+//
+// *Request messages are sent to the process from Kapacitor.
+// *Response messages are sent to Kapacitor from the process.
+//
+// While there is an obvious request/response structure for communicating,
+// there is a loose coupling between request and response.
+// Meaning that ordering or synchronizing STDIN and STDOUT in anyway
+// is not necessary.
+// For example if Kapacitor requests a state snapshot and the
+// process is in the middle of writing a previous response or
+// data points those can continue. Eventually Kapacitor will receive
+// the state snapshot response and act accordingly.
+//
+// A KeepAliveRequest/KeepAliveResponse system is used to ensure that 
+// the process is responsive. Every time that a KeepAliveRequest is sent
+// a KeepAliveResponse must be returned within a timeout. If too many heartbeats
```

Comment:
I would modify this so that any message received by Kapacitor from STDOUT should reset the timer, because the process is obviously still alive?